sunxspex/tests/test_thermal.py

```diff
@@ -0,0 +1,149 @@
+        1.47199e-10, 5.36175e-11, 1.96076e-11, 7.20292e-12, 2.64275e-12, 9.68337e-13, 3.54305e-13,
+        1.30357e-13, 4.81556e-14, 1.77739e-14, 6.55418e-15, 2.41451e-15, 8.88565e-16, 3.26635e-16,
+        1.20928e-16, 4.49286e-17, 1.66830e-17, 6.19118e-18, 2.29618e-18, 8.51055e-19, 3.15220e-19,
+        1.16669e-19, 4.31491e-20, 1.59455e-20, 5.90236e-21, 2.20614e-21, 8.24339e-22, 3.07924e-22,
+        1.14983e-22] * SSW_INTENSITY_UNIT
+    return inputs, ssw_output
+
+
+def chianti_kev_lines_simple():
+    """Define expected thermal line spectrum as returned by SSW routine chianti_kev_lines.
+
+    chianti_cont_kev is the standard routine used to calculate a solar X-ray line spectrum
+    in SSW.  It is used by f_vth.pro.
+    The output defined here uses a energy bins from 3-28 keV with a bin width of 0.5 keV,
+    a temperature of 6 MK, a emission measure of 1e44 cm^-3, an observer distance of 1AU,
+    default ('sun_coronal') abundances, and default relative abundances.
+    The SSW output can be reproduced in SSW/IDL with the following code:
+
+    IDL> energy_in = fltarr(2, 50)
+    IDL> energy_in[0, *] = findgen(50) * 0.5 + 3
+    IDL> energy_in[1, *] = energy_in[0, *] + 0.5
+    IDL> temp = 6.
+    IDL> flux = chianti_kev_lines(temp, energy_in, /kev, /earth)
+    """
+    energy_edges = np.arange(3, 28.5, 0.5) * u.keV
+    temperature = 6 * u.MK
+    emission_measure = 1e44 / u.cm**3
+    abundance_type = "sun_coronal"
+    relative_abundances = None
+    observer_distance = (1 * u.AU).to(u.cm)
+    inputs = (energy_edges, temperature, emission_measure, abundance_type, relative_abundances,
+              observer_distance)
+    ssw_output = [
+        0.073248975,   0.014247916,   0.0044440511,  0.00067793718, 2.3333176e-05, 2.5751346e-10,
+        3.4042361e-05, 2.1403499e-05, 5.0370664e-07, 8.3715751e-12, 2.7737142e-12, 1.5496721e-13,
+        1.9522280e-17, 1.3281716e-20, 1.0493879e-21, 0.0000000,     0.0000000,     0.0000000,
+        0.0000000,     0.0000000,     0.0000000,     0.0000000,     0.0000000,     0.0000000,
+        0.0000000,     0.0000000,     0.0000000,     0.0000000,     0.0000000,     0.0000000,
+        0.0000000,     0.0000000,     0.0000000,     0.0000000,     0.0000000,     0.0000000,
+        0.0000000,     0.0000000,     0.0000000,     0.0000000,     0.0000000,     0.0000000,
+        0.0000000,     0.0000000,     0.0000000,     0.0000000,     0.0000000,     0.0000000,
+        0.0000000,     0.0000000] * SSW_INTENSITY_UNIT
+    return inputs, ssw_output
+
+
+@pytest.mark.parametrize("ssw",
+                            (
+                                (fvth_simple,)
+                            ))
+def test_thermal_emission_against_ssw(ssw):
+    input_args, expected = ssw()
+    output = thermal_emission(*input_args)
+    expected_value = expected.to_value(output.unit)
+    np.testing.assert_allclose(output.value, expected_value, rtol=0.03)
```

Comment:
Yes that's right.  The reason `atol=1e-30`(absolute tolerance) is used in the `line_emission` test is because `rtol` (relative tolerance) doesn't work when the expected answer is `0.`.

Reply:
Any idea why they are that far off, I would have expected a lot less e.g. 1e-6 ?

Reply:
1e-30 is 24 orders of magnitude smaller than 1e-6.

Reply:
That's right.